File: 858-mirror-reflection/858-mirror-reflection.py
class Solution:
    def mirrorReflection(self, p: int, q: int) -> int:
        while(q%2==0 and p%2==0):
            p/=2;
            q/=2;
        if(q%2==0 and p%2!=0): return 0;
        if(q%2!=0 and p%2==0): return 2;
        if(q%2!=0 and p%2!=0): return 1;
        return -1;
        

File: 858-mirror-reflection/858-mirror-reflection.cpp
class Solution {
public:
    int mirrorReflection(int p, int q) {
      // int ext=q,reflection=p;
        while(q%2==0&&p%2==0)
            q/=2,p/=2;
        if(q%2==0&&p%2!=0) return 0;
        if(q%2!=0&&p%2==0) return 2;
        if(q%2!=0&&p%2!=0) return 1;
        
        return -1;
    }
};